src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateClone = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateClone = Object.assign(stateClone, action.extraData);
```

Comment:
```suggestion
        Object.assign(stateClone, action.extraData);
```
Reassignment is not necessary

Reply:
++